File: Benchmarking/src/gen_1115_5.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    s0 p0;
    std::vector<std::vector<s0>> p1;
    s1(s0 v0, std::vector<std::vector<s0>> v1): p0(v0), p1(v1) { }
  };
  struct s5 {
    std::vector<s0> p0;
    s0 p1;
    s5(std::vector<s0> v0, s0 v1): p0(v0), p1(v1) { }
  };
  struct s8 {
    std::vector<s0> p0;
    std::vector<s5> p1;
    s8(std::vector<s0> v0, std::vector<s5> v1): p0(v0), p1(v1) { }
  };
  double f40(const double &v0) {
    double v4 = v0;
    double v7 = v4;
    v7 = v0;
    double v6 = v0;
    double v5 = v0;
    const double v2 = v6 + v7;
    double v11 = v6;
    const std::vector<double> v12 { v11, v5 };
    double v14 = v2;
    std::vector<double> v21 = v12;
    std::vector<double> v25 = v21;
    const double v41 = v25[1];
    v25[1] = v14;
    return v41;
  }
  __attribute__((noinline))
  double f0(const s1 &v0, const s8 &v1, const double &v2) {
    const s0 v9 = v0.p0;
    const std::vector<std::vector<double>> v60 = v9.p0;
    double v31 = v2;
    const std::vector<double> v42 = v60[0];
    const double v45 = f40(v31);
    const double v82 = v42[0];
    std::vector<double> v32 = v42;
    v32[0] = v82;
    std::vector<double> v55 = v32;
    const double v87 = v55[0];
    const double v86 = f40(v45);
    v32[0] = v86;
    v32[0] = v2;
    return v87;
  }
  int main() {
    s1 v0({ { { { 0.0 }, { 1.0 } }, { { 2.0 }, { 3.0 }, { 4.0 } } }, { { { { { 5.0 }, { 6.0 } }, { { 7.0 }, { 8.0 }, { 9.0 } } } }, { { { { 10.0 }, { 11.0 } }, { { 12.0 }, { 13.0 }, { 14.0 } } } }, { { { { 15.0 }, { 16.0 } }, { { 17.0 }, { 18.0 }, { 19.0 } } } } } });
    s8 v1({ { { { { 20.0 }, { 21.0 } }, { { 22.0 }, { 23.0 }, { 24.0 } } } }, { { { { { { 25.0 }, { 26.0 } }, { { 27.0 }, { 28.0 }, { 29.0 } } }, { { { 30.0 }, { 31.0 } }, { { 32.0 }, { 33.0 }, { 34.0 } } } }, { { { 35.0 }, { 36.0 } }, { { 37.0 }, { 38.0 }, { 39.0 } } } }, { { { { { 40.0 }, { 41.0 } }, { { 42.0 }, { 43.0 }, { 44.0 } } }, { { { 45.0 }, { 46.0 } }, { { 47.0 }, { 48.0 }, { 49.0 } } } }, { { { 50.0 }, { 51.0 } }, { { 52.0 }, { 53.0 }, { 54.0 } } } } } });
    double v2(55.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen_1115_5.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: s0
    var p1: [[s0]]
  }
  struct s5 {
    var p0: [s0]
    var p1: s0
  }
  struct s8 {
    var p0: [s0]
    var p1: [s5]
  }
  func f40(_ v0: Double) -> Double {
    var v4: Double = v0
    var v7: Double = v4
    v7 = v0
    var v6: Double = v0
    var v5: Double = v0
    let v2: Double = v6 + v7
    var v11: Double = v6
    let v12: [Double] = [v11, v5]
    var v14: Double = v2
    var v21: [Double] = v12
    var v25: [Double] = v21
    let v41: Double = v25[1]
    v25[1] = v14
    return v41
  }
  @inline(never)
  func f0(_ v0: s1, _ v1: s8, _ v2: Double) -> Double {
    let v9: s0 = v0.p0
    let v60: [[Double]] = v9.p0
    var v31: Double = v2
    let v42: [Double] = v60[0]
    let v45: Double = f40(v31)
    let v82: Double = v42[0]
    var v32: [Double] = v42
    v32[0] = v82
    var v55: [Double] = v32
    let v87: Double = v55[0]
    let v86: Double = f40(v45)
    v32[0] = v86
    v32[0] = v2
    return v87
  }
  func benchmark() {
    let v0: s1 = s1(p0: s0(p0: [[0.0], [1.0]], p1: [[2.0], [3.0], [4.0]]), p1: [[s0(p0: [[5.0], [6.0]], p1: [[7.0], [8.0], [9.0]])], [s0(p0: [[10.0], [11.0]], p1: [[12.0], [13.0], [14.0]])], [s0(p0: [[15.0], [16.0]], p1: [[17.0], [18.0], [19.0]])]])
    let v1: s8 = s8(p0: [s0(p0: [[20.0], [21.0]], p1: [[22.0], [23.0], [24.0]])], p1: [s5(p0: [s0(p0: [[25.0], [26.0]], p1: [[27.0], [28.0], [29.0]]), s0(p0: [[30.0], [31.0]], p1: [[32.0], [33.0], [34.0]])], p1: s0(p0: [[35.0], [36.0]], p1: [[37.0], [38.0], [39.0]])), s5(p0: [s0(p0: [[40.0], [41.0]], p1: [[42.0], [43.0], [44.0]]), s0(p0: [[45.0], [46.0]], p1: [[47.0], [48.0], [49.0]])], p1: s0(p0: [[50.0], [51.0]], p1: [[52.0], [53.0], [54.0]]))])
    let v2: Double = 55.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 55.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
